src/messages/tests.rs

```diff
@@ -0,0 +1,410 @@
+use rand::thread_rng;
+
+use super::*;
+
+use crate::messages::validate::{MsgErr, Validate};
+
+use crate::frost;
+
+use serde_json;
+use std::convert::TryFrom;
+
+#[test]
+fn validate_version() {
+    // A version number that we expect to be always invalid
+    const INVALID_VERSION: u8 = u8::MAX;
+
+    let signer1 = ParticipantId::Signer(0);
+    let dealer = ParticipantId::Dealer;
+
+    let validate = Validate::validate(&Header {
+        version: MsgVersion(INVALID_VERSION),
+        sender: dealer.clone(),
+        receiver: signer1.clone(),
+    })
+    .err()
+    .expect("an error");
+
+    assert_eq!(validate, MsgErr::WrongVersion);
```

Comment:
This test works, but here's a slightly simpler way to write it:
```suggestion
    let validate = Validate::validate(&Header {
        version: MsgVersion(INVALID_VERSION),
        sender: dealer.clone(),
        receiver: signer1.clone(),
    });
    assert!(matches!(validate, Err(MsgErr::WrongVersion)));
```

There is no need to change any tests use `matches!` - I'm just letting you know for future tests.

Reply:
Yea, i implemented this suggestions in all the tests https://github.com/oxarbitrage/redjubjub/pull/2/commits/30b72b7f9caa1e0f1f6f1b983b67824cb48d23ce as i am repeating a lot. There is more room to improve the test code further.